shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.mm

```diff
@@ -335,6 +341,8 @@
                                                                  FlutterSemanticsAction action,
                                                                  const std::vector<uint8_t>& data) {
   NSCAssert(flutter_engine_, @"Flutter engine should not be deallocated");
+  NSCAssert(flutter_engine_.viewController.viewLoaded && flutter_engine_.viewController.view.window,
+            @"A headless engine should not receive accessibility actions");
```

Comment:
This log message is misleading as written; `window` being null does not necessarily indicate that the engine is headless.